pkg/k8s/watchers/endpointsynchronizer.go

```diff
@@ -246,6 +249,14 @@ func (epSync *EndpointSynchronizer) RunK8sCiliumEndpointSync(e *endpoint.Endpoin
 				// If it fails it means the test from the previous patch failed
 				// so we can safely replace this node in the CNP status.
 				replaceCEPStatus := []k8s.JSONPatch{
+					// If the stored UID matches the one in the ciliumendpoint then
+					// this first patch is a no-op, otherwise the entire patch will
+					// not be applied as uid is immutable.
+					{
+						OP:    "replace",
```

Comment:
Why don't we use the `OP: "test"` that tests for the equalness of `/metadata/uid` and `e.GetCiliumEndpointUID()`?

Reply:
Yeah, that makes sense 👍